src/APIBasedAuth.ts

```diff
@@ -0,0 +1,192 @@
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse
+} from 'axios';
+
+/* istanbul ignore next */
+const formatAxiosError = <ErrorResponse = any>(
+  error: AxiosError<ErrorResponse> | unknown
+) => {
+  if (!axios.isAxiosError(error)) {
+    throw new AxiosError(String(error));
+  }
+  throw error.response?.data;
+};
+
+class APIBasedAuth {
+  private client: AxiosInstance;
+  private clientOptions: APIBasedAuth.Config;
+  private session?: APIBasedAuth.Session;
+  private tokenData?: APIBasedAuth.TokenData;
+
+  constructor(options: APIBasedAuth.Config) {
+    const required = ['baseURL', 'clientId'];
+    for (const param of required) {
+      // eslint-disable-next-line security/detect-object-injection
+      if (!options[param]) {
+        throw new Error(`APIBasedAuth ctor param ${param} is required`);
```

Comment:
What is `ctor`?

Reply:
no clue, it was in there for the LOAuth so didn't rename it

Reply:
+1 for removing it.